app/Support/Shortcode/Shortcode.php

```diff
@@ -40,6 +42,29 @@ public static function render(string $input, array $options = []): string
         return (new Shortcode())->parse($input, $options);
     }
 
+    public static function convertUserShortcodesToUseIds(string $input): string
+    {
+        // Extract all usernames from the payload. We want to make a single
+        // query so someone doesn't inadvertently slam the database.
+        preg_match_all('/\[user=(.*?)\]/', $input, $matches);
+        $usernames = $matches[1];
+
+        if (empty($usernames)) {
+            return $input;
+        }
+
+        // Fetch all users by username in a single query.
+        $users = User::whereIn('User', $usernames)->get()->keyBy('User');
```

Comment:
Soft-deleted users aren't being converted, but appear to be functional.
```
| Ask [user=Ansku].         |
```
Also doesn't seem to be converting incorrectly cased users:
```
| Ask [user=14863].         |
| Ask [user=Televandalist]. |
```

Reply:
Good catches - both fixed in latest.

Reply:
Deleted user is being converted to an int, but not converted back.
![image](https://github.com/user-attachments/assets/1ffefa2f-748b-4bd9-82aa-b3c4bcf0a5b6)

Reply:
Resolved in latest.